File: SymptomCheckFrontend/src/app/models/availability-health-dto.ts


export interface AvailabilityHealthDto {
  doctorId: string;
  daysOfWeek: string[];
  startTime: string;
  endTime: string;
  name?: string;
  description?: string;
  imageUrl?: string;
  durationMinutes?: number;
  price?: number;
}


export enum DayOfWeek {
  MONDAY = 'MONDAY',
  TUESDAY = 'TUESDAY',
  WEDNESDAY = 'WEDNESDAY',
  THURSDAY = 'THURSDAY',
  FRIDAY = 'FRIDAY',
  SATURDAY = 'SATURDAY',
  SUNDAY = 'SUNDAY',
}

File: SymptomCheck-backend/doctorservice/src/main/java/com/symptomcheck/doctorservice/dtos/AvailabilityHealthDto.java
package com.symptomcheck.doctorservice.dtos;

import jakarta.persistence.Column;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.DayOfWeek;
import java.time.LocalTime;
import java.util.List;
import java.util.UUID;

@Data

public class AvailabilityHealthDto {
    private UUID doctorId;
    @NotNull
    private List<DayOfWeek> daysOfWeek;
    @NotNull
    private LocalTime startTime;
    @NotNull
    private LocalTime endTime;
    private String name;
    @Column(length = 1000)
    private String description;
    private String category;
    private String imageUrl;
    private Integer durationMinutes;
    private Double price;
}
